src/bg/execute.js

```diff
@@ -30,27 +30,48 @@ function executeUserscriptOnNavigation(details) {
     });
   }
 
-  // TODO: Optional feature.
-  updateBadgeByDetails(details);
+  // TODO: User configurable feature.
+  updateScriptStatsByDetails(details);
 }
 
 
-function updateBadgeByDetails(details) {
+function updateScriptStatsByDetails(details) {
   if (false === getGlobalEnabled()) return;
+  if (0 !== details.frameId) return;
 
-  let userScriptIterator = UserScriptRegistry.scriptsToRunAt(details.url);
-  let count = 0;
-  for (let userScript of userScriptIterator) count++;
-  if (count) {
+  const enabled = true;  // TODO: query ('EnabledQuery' not yet available).
+  let count = [[0, 0], [0, 0]];
+  if (enabled) {
+    for (let i = 0; i < 2; i++) {
+      for (let j = 0; j < 2; j++) {
```

Comment:
Assuming I'm understanding this right then what you're calculating here is
```
(scripts active that run on url) / (total scripts that run on url)
```
and
```
(scripts active) / (total scripts)
```

The way it's written is awfully confusing with the multidimensional count array. It would probably be much better to create a new function within the registry that will tally everything and return an object with keys for each count. Since the registry has all the data it only needs to iterate once.

Reply:
Actually, the registry yields the userscript object directly. If you want you can still do it outside the registry. Just ask for "all" scripts and then do the checks.
```javascript
for (scripts) {
  let runOn = script.runsAt(url);
  runActive += (runOn && script.enabled) ? 1 : 0;
  runTotal += runOn ? 1 : 0;
  totalActive += script.enabled ? 1 : 0;
  total += 1
}
```

Reply:
The registry is (for sure) a better place for counting this data. I don't like the counting (we don't neet the data, just the count), I don't really like that array, but I didn't like plenty of different similar filenames either. I wanted to avoid a change to the registry, but I should count there! This will not be immediately be merged, so there's time to do that.

The association is here:
```
        const url = (i ? details.url : null);
        const all = !j;
```
First parameter `i` is `detected[1]` vs. `all[0]` scripts, second parameter is `activated[1]` vs. `total[0]` scripts. `all` means `detected`+`ignored`, `total` means `activated`+`deactivated`.

Thus the first line contains the `activated` scripts: running vs. idle (but active) scripts.
The second line contains all the scripts: running vs. all (active or not) scripts.
It's always from the viewpoint "how many scripts are running out of the available set of scripts".

It would also make sense to transpose that order, but I liked this one.

------------------------------------------------------------------------

My example (before I nuked my database) was:

In the GM menu, I had 6 scripts `for this tab`, 2 of them were active, the other 4 were greyed.
I had 20 `other` scripts, 13 of them were active, the other 7 were greyed.

The output was
```
Greasemonkey
2 / 15 active
6 / 26 total
```
(you could alternatively give 2/6 and 15/26 as output)

Reply:
Your little `for` loop should also work fine, except that I would write
```
  const runOn = script.runsAt(url);
  if (runOn && script.enabled) runActive++;
  if (runOn) runTotal++;
  if (script.enabled) totalActive++;
  total++;
```

Reply:
Correcting your initial assumption:

> Assuming I'm understanding this right then what you're calculating here is
> 
> `(active scripts that run on url) / (all active scripts)`
> 
> and
> 
> `(total scripts that run on url) / (total scripts)`